graphql_compiler/cost_estimation/int_value_conversion.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+"""Int value conversion.
+
+This module is a utility for reasoning about intervals when computing filter selectivities,
+and generating parameters for pagination. Since integers are the easiest type to deal with
+in this context, when we encounter a different type we represent it as an int, do all the
+computation in the integer domain, and transfer the computation back into the original domain.
+
+In order to be able to reason about value intervals and successor/predecessor values, we
+make sure these mappings to integers are increasing bijective functions.
+
+This kind of mapping is easy to do for int, uuid and datetime types, but not possible for other
+types, like string. When the need for other types arises, the precise interface for range
+reasoning can be defined and implemented separately for each type.
+"""
+import datetime
+from uuid import UUID
+
+import pytz
+
+from .helpers import is_date_field_type, is_datetime_field_type, is_int_field_type, is_uuid4_type
+
+
+# UUIDs are defined in RFC-4122 as a 128-bit identifier. This means that the minimum UUID value
+# (represented as a natural number) is 0, and the maximal value is 2^128-1.
+MIN_UUID_INT = 0
+MAX_UUID_INT = 2**128 - 1
+
+
+DATETIME_1970 = datetime.datetime(1970, 1, 1, tzinfo=pytz.utc)
```

Comment:
Consider calling this `DATETIME_EPOCH_UTC`, 1970-01-01 is canonically the epoch: https://en.wikipedia.org/wiki/Epoch_(computing)